### PR TITLE
Add methods to stop&start probes, cleanup gRPC service a bit.

### PR DESCRIPTION
**Pull request by vtsingaras**
* Add methods to start and stop probes (for use by lameducking)
* Have gRPC method implementations call these methods
* Add support for lameducking when host maintenance happens

### CODE REVIEW

**Comment by manugarg** [comment]
Thanks for the PR, @vtsingaras! Can you please summarize your changes.

**Review by manugarg — COMMENTED** [review]
Hey Vyron, Please see my inline comment. Also, this PR needs a more details in the summary. One line summary doesn't explain all the changes being done :(

Are you trying to add new gRPC API methods to stop/start probes? If that's the case, you'll need to modify the service definition (in protobuf) too.

File: prober/prober.go [hunk 1]
```diff
@@ -241,29 +254,77 @@ func (pr *Prober) Start(ctx context.Context) {
...
+	for name := range pr.Probes {
+		pr._stopProbeWithNoLock(name)
+	}
+}
+
+func (pr *Prober) getProbes() map[string]*probes.ProbeInfo {
+	pr.mu.Lock()
+	defer pr.mu.Unlock()
+
+	var copier = cpy.New(
```
[the remark this answers is not in the document]
Comment (vtsingaras) [reply prober/prober.go:322]: This is essentially doing this for every item in the map (we are returning here a map[string]ProbeInfo not a single ProbeInfo)

**Comment by vtsingaras** [comment]
@manugarg PR has been updated

**Comment by manugarg** [comment]
Couple of big items before I do more in depth review:

* Can you please generate the protobuf code again with protoc version 21.5. Following script should do this automatically if protoc is not in the path: https://github.com/cloudprober/cloudprober/blob/master/tools/gen_pb_go.sh. As it is, it's super hard to review this PR.
* Can you please remove the dependency on `cpy`. Here we know the concrete type and can do the copy ourselves -- it's better to not add a dependency if it's not adding much value. Also, I didn't understand why are you adding the getProbes() method, and returning a full data structure, if you're going to use it only in ListProbes method.

**Comment by manugarg** [comment]
Closing this now.

**Pull request closed without merging**